src/lib/y2network/autoinst_profile/section_with_attributes.rb

```diff
@@ -0,0 +1,192 @@
+# encoding: utf-8
+
+# Copyright (c) [2019] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+
+module Y2Network
+  module AutoinstProfile
+    # Abstract base class for some AutoYaST profile sections
+    #
+    # @todo This class is supposed to live in autoyast2 or yast2. When done,
+    #   please, adapt yast2-storage-ng accordingly.
+    class SectionWithAttributes
+      include Yast::Logger
+
+      class << self
+        # Description of the attributes in the section.
+        #
+        # To be defined by each subclass. Each entry contains a hash with the
+        # mandatory key :name and an optional key :xml_name.
```

Comment:
```suggestion
        # mandatory key :name and an optional key :xml_name; the values are symbols.
```

Reply:
So far I see no use of `:xml_name`, and the instantiated `attributes` methods are seemingly redundant with `name:` all over. Is this needed?

Reply:
well, for now it is just copy from yast-storage-ng and hopefully this will be moved to autoyast or yast2 itself as it is generic class.

Reply:
Oh! Then add a comment "Copied from yast-storage-ng, TODO deduplicate" otherwise they will diverge.